File: Domain/Entities/ChatRoom.cs
using Domain.Common;

namespace Domain.Entities
{
    /// <summary>
    /// Entity that represents a chat room.
    /// </summary>
    public class ChatRoom : BaseAuditableEntity
    {
        public Guid? ChatRoomTypeId { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }

        public virtual ChatRoomType? ChatRoomType { get; set; }

        public virtual ICollection<UserChatRoom> ChatRoomUsers { get; set; } = new HashSet<UserChatRoom>();
        public virtual ICollection<UserChat> UserChats { get; set; } = new HashSet<UserChat>();
    }
}

File: client-app/src/app/models/chatRoom.ts
import { UserChat } from "./userChat";
import { UserChatRoom } from "./userChatRoom";

export interface ChatRoom { 
    id: string,
    name?: string,
    description?: string,

    chats: UserChat[],
    users: UserChatRoom[]
}

export interface ChatRoomDto {
    id: string,
    displayName?: string
}